src/ui/public/doc_table/components/table_header.js

```diff
@@ -94,8 +94,18 @@ module.directive('kbnTableHeader', function (shortDotsFilter) {
           : 'asc'
         );
 
+        $scope.sortText = (
+          (columnName === currentColumnName && currentDirection === 'asc')
+          ? 'Button: Sort time ascending'
+          : 'Button: Sort time descending'
+        );
+
         $scope.onChangeSortOrder(columnName, newDirection);
       };
+
+      $scope.sortText = $scope.sortOrder[1] === 'asc'
+        ? 'Button: Sort time descending'
+        : 'Button: Sort time ascending';
```

Comment:
Great idea! But I think we'll need to extract this out into its own scope method, and tweak it to use a dynamic column name. Something like this (haven't tested that this works):

```html
<button aria-label="{{ getAriaLabelForColumn('time') }}">
```

```javascript
$scope.getAriaLabelForColumn = function getAriaLabelForColumn(name) {
  const [currentColumnName, currentDirection = 'asc'] = $scope.sortOrder;
  if(name === currentColumnName && currentDirection === 'asc') {
    return `Button: Sort ${name} ascending`;
  }

  return `Button: Sort ${name} descending`;
}
```